src/plugins/data_source_management/public/components/edit_data_source/components/edit_form/edit_data_source_form.tsx

```diff
@@ -69,7 +69,7 @@ export interface EditDataSourceState {
 
 export class EditDataSourceForm extends React.Component<EditDataSourceProps, EditDataSourceState> {
   static contextType = contextType;
-  public readonly context!: DataSourceManagementContextValue;
+  declare context: React.ContextType<typeof contextType>;
```

Comment:
Many thanks @ruanyl  for giving me such great suggestions and feedback! Another problem I'm getting is test failing with the below error. I'm unsure what that means but from some research, it seems like I should make some changes to the config file. For context, I added declare to get rid of the type error. If this is the right approach, would changing the config file be the right approach?

potential fix: https://stackoverflow.com/questions/70690029/typescript-declare-fields-must-first-be-transformed-by-babel-plugin-transform 

failed test: https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/7256678119/job/19769311750?pr=5546 

![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/70486227/fa679ea6-c97f-42f1-87c5-4f8f2258d8d3)

Reply:
I did some investigation, but couldn't find a straight-forward solution, I might leave it as it was for now.

Reply:
So instead of changing the config file, I might just not trying to fix the type error for now.

Reply:
Got it, I already reverted this change in a commit about an hour ago. 
Thanks!